lib/std/fs.zig

```diff
@@ -2238,24 +2238,27 @@ pub const Dir = struct {
         return file.stat();
     }
 
-    pub const StatFileError = File.OpenError || StatError;
+    pub const StatFileError = File.OpenError || File.StatError || os.FStatAtError;
 
-    // TODO: simplify this once windows get a proper fstatat implementation.
-    pub fn statFile(self: Dir, sub_path: []const u8) if (builtin.os.tag == .windows) StatFileError!Stat else os.FStatAtError!Stat {
+    // statFile provides info on a file (File.Stat) for any file in the opened directory,
+    // with a single syscall (fstatat), except on Windows.
+    // Currently on Windows, files are opened then closed (implying several syscalls, unfortunately).
+    // Symlinks are not followed on linux, haiku, solaris and *BSDs.
+    // Other OSs have a default behavior (they currently lack an os.AT.SYMLINK_NOFOLLOW flag).
```

Comment:
Both seem rather true. Since `os.AT.SYMLINK_NOFOLLOW` isn't implemented everywhere, we are lacking a generic flag to control if symlinks should be followed.

In any case, I don't think it is a problem of the OSs per se, more a problem of the Zig standard library. Said otherwise, I don't think there is an OS mandating applications to follow symlinks.

Thus, this is related to `TODO: correct flags not yet implemented` because since we don't have the `os.AT.SYMLINK_NOFOLLOW` flag, we currently have no way to express "don't follow symlinks", at least with a simple flag (and to my knowledge).